src/tools/vdorecover/vdorecover

```diff
@@ -64,8 +64,11 @@ _fstrimAndPrompt(){
 	local KEEPGOING=true
 
 	while $KEEPGOING; do
-		fstrim $MOUNT_POINT || true
-		FSOUT=$(echo $?)
+		# If we weren't provided a $MOUNT_POINT, then don't try to run fstrim
+		if [[ -n $MOUNT_POINT ]]; then
+			fstrim $MOUNT_POINT || true
+			FSOUT=$(echo $?)
+		fi
```

Comment:
if FSOUT is only  conditionally set, the test of it further down needs fixing. You could initialize it to 0 first.

But "some_command || true" will always have an exit status of 0, so the FSOUT bits could just be eliminated if you want to ignore fstrim errors.

Reply:
I thought about that.  I'd have to set FSOUT to something non-zero since the check is validating a non-zero status.

I think initializing it to non-zero (e.g. 255?)  to start with would make more sense.  Particularly because the only cases where it's relevant are where this is running on a VDO manager based system and the filesystem is mounted.

The `|| true` aspect of it is also necessary, though since the script overall is using `set -e`.

Reply:
Using ||true means the exit status $? will be from the end of the sequence, namely the "true" command. If you need to examine a non-zero exit status from a command, get rid of -e, or put it in an "if" test or "||"/"&&"  (which -e will let fail without killing the script).

Reply:
I'm assuming you mean spelling it like this?  That's how I'd normally have achieved this in other contexts.

```
fstrim $MOUNTPOINT || exitVal=$? && exitVal=0
```

As much as I'd like to drop the check altogether, I'm sure it was put in there for a reason.  Assuming you're in agreement with my suggested spelling, then I will update accordingly and re-test.

Reply:
I've gone ahead and at least posted with the new spelling after testing it.  It works fine with both LVMVDO and vdo manager based volumes.